File: Data Structures/Array/Multi-dimensional/Python3/Transpose.py
n, m = map(int, input().split())
mat = [
    [int(x) for x in input().split()]
    for i in range(n)
]
ans = [[0 for _ in range(n)] for _ in range(m)]

# print(ans)
for i in range(n):
    for j in range(m):
        ans[j][i] = mat[i][j]


for i in ans:
    print(*i)

File: Data Structures/Array/Multi-dimensional/C++/Transpose.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n, m;
    cin >> n >> m;

    // Input matrix
    vector<vector<int>> mat(n, vector<int>(m));
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < m; ++j) {
            cin >> mat[i][j];
        }
    }

    // Transpose matrix
    vector<vector<int>> ans(m, vector<int>(n));
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < m; ++j) {
            ans[j][i] = mat[i][j];
        }
    }

    // Print transposed matrix
    for (int i = 0; i < m; ++i) {
        for (int j = 0; j < n; ++j) {
            cout << ans[i][j] << " ";
        }
        cout << endl;
    }

    return 0;
}
